hyper/http11/connection.py

```diff
@@ -75,6 +79,19 @@ def __init__(self, host, port=None, secure=None, ssl_context=None,
         self.ssl_context = ssl_context
         self._sock = None
 
+        # Setup proxy details if applicable.
+        if proxy_host:
+            if proxy_port is None:
+                try:
+                    self.proxy_host, self.proxy_port = proxy_host.split(':')
+                    self.proxy_port = int(self.proxy_port)
+                except ValueError:
+                    self.proxy_host, self.proxy_port = proxy_host, 8080
+            else:
+                self.proxy_host, self.proxy_port = proxy_host, proxy_port
+        else:
+            self.proxy_host = None
```

Comment:
Can we `None` out `self.proxy_port` here as well? I hate having attributes that only *sometimes* exist.